nexus/src/external_api/http_entrypoints.rs

```diff
@@ -1106,6 +1120,249 @@ async fn instance_disks_detach(
+        let images = nexus
+            .list_images(
+                &opctx,
+                &data_page_params_for(&rqctx, &query)?
+                    .map_name(|n| Name::ref_cast(n)),
+            )
+            .await?
+            .into_iter()
+            .map(|d| d.into())
+            .collect();
+        Ok(HttpResponseOk(ScanByName::results_page(&query, images)?))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Create a global image.
+#[endpoint {
+    method = POST,
+    path = "/images",
+    tags = ["images"]
+}]
+async fn images_post(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    new_image: TypedBody<params::ImageCreate>,
+) -> Result<HttpResponseCreated<Image>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let new_image_params = &new_image.into_inner();
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let image = nexus.create_image(&opctx, &new_image_params).await?;
+        Ok(HttpResponseCreated(image.into()))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Path parameters for Image requests
+#[derive(Deserialize, JsonSchema)]
+struct GlobalImagePathParam {
+    image_name: Name,
+}
+
+/// Get a global image.
+#[endpoint {
+    method = GET,
+    path = "/images/{image_name}",
+    tags = ["images"],
+}]
+async fn images_get_image(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<GlobalImagePathParam>,
+) -> Result<HttpResponseOk<Image>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let image_name = &path.image_name;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let image = nexus.image_fetch(&opctx, &image_name).await?;
+        Ok(HttpResponseOk(image.into()))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Delete a global image.
+#[endpoint {
+    method = DELETE,
+    path = "/images/{image_name}",
+    tags = ["images"],
+}]
+async fn images_delete_image(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<GlobalImagePathParam>,
+) -> Result<HttpResponseDeleted, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let path = path_params.into_inner();
+    let image_name = &path.image_name;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        nexus.delete_image(&opctx, &image_name).await?;
+        Ok(HttpResponseDeleted())
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// List images in a project.
+#[endpoint {
+    method = GET,
+    path = "/organizations/{organization_name}/projects/{project_name}/images",
+    tags = ["images"],
```

Comment:
```suggestion
    tags = ["projects"],
```

Reply:
Is there documentation on what tag should be associated with what endpoint? I think I might not be understanding this fully. I thought that an operation acting on:

```
path = "/organizations/{organization_name}/projects"
```

Would be tagged with `projects`, but an operation acting on:

```
path = "/organizations/{organization_name}/projects/{project_name}/foobar"
```

Would be tagged with `foobar(s)`.

Are *both* supposed to be tagged with `projects`? If we keep nesting, is it just `projects` all the way down?